diff --git a/countUp-jquery.js b/countUp-jquery.js
index e086da0..57f5629 100644
--- a/countUp-jquery.js
+++ b/countUp-jquery.js
@@ -24,9 +24,8 @@
         	return;
         } 
 
-        this.each(function(i) {
-        	// var CountUp = function(target, startVal, endVal, decimals, duration, options) {
-        	var countUp = new CountUp(this[i], defaults.startVal, defaults.endVal, defaults.decimals, defaults.duration, defaults.options);
+        this.each(function(i, elem) {
+        	var countUp = new CountUp(elem, defaults.startVal, defaults.endVal, defaults.decimals, defaults.duration, defaults.options);
 
         	countUp.start();
         });
